frontend/ui/elements/screen_dpi_menu_table.lua

```diff
@@ -20,6 +20,28 @@ local function setDPI(_dpi)
     Device:setScreenDPI(_dpi)
 end
 
+local function spinWidgetSetDPI()
+    local SpinWidget = require("ui/widget/spinwidget")
+    local UIManager = require("ui/uimanager")
+    local items = SpinWidget:new{
+        width = Screen:getWidth() * 0.6,
+        value = custom() or dpi(),
+        value_min = 90,
+        value_max = 900,
+        value_step = 10,
+        value_hold_step = 50,
+        ok_text = _("Set DPI"),
+        title_text = _("Set custom screen DPI"),
+        callback = function(spin)
+            G_reader_settings:saveSetting("custom_screen_dpi", spin.value)
+            setDPI(spin.value)
+            local Event = require("ui/event")
+            UIManager:broadcastEvent(Event:new("UpdateTouchMenu"))
```

Comment:
May be you don't need to add this onUpdateTouchMenu and use a broadcast.
A callback called from a menu item gets passed the touchmenu_instance, and you can call updateItems on it.
For example (even if it's a bit different from your situation):
https://github.com/koreader/koreader/blob/e2961097e7b054d97928b6d188353335372b0db6/frontend/apps/reader/modules/readerrolling.lua#L372-L387

Reply:
I didn't know about that trick :)
Thanks, it works :)